api/v1alpha/src/lib/metaManager.js

```diff
@@ -94,107 +99,141 @@ async function performDatasetMetadataUpdate(projectId, datasetId, accounts) {
             console.error(`Failed to set metadata for dataset '${datasetId}' with error '${err}' and payload: ${JSON.stringify(metadata)}`);
             throw err;
         }
+    } else {
+        console.info(`Metadata is already up to date for dataset: '${datasetId}'`);
     }
 
+    console.log(`End metadata update for dataset: ${datasetId}`);
     return isDirty;
 }
 
 /**
  * @param  {} projectId
- * @param  {} policyIds
- * @param  {} datasetIds
+ * @param  {} datasetId
+ * @param  {} tableId
+ * @param  {} accounts
  */
-async function performMetadataUpdate(projectId, policyIds, datasetIds) {
-    // Remove the datasetIds parameter, and get this ourselves based on the last version of the policyId.
-    // For each policyId get the prior record and update using that.
-    
-    console.log(`performMetadataUpdate called for policyIds: ${JSON.stringify(policyIds)} and datasetIds: ${JSON.stringify(datasetIds)}`);
-    let filter = "";
-    if (policyIds && policyIds.length > 0 && datasetIds && datasetIds.length > 0) {
-        filter = "(cp.policyId IN UNNEST(@policy_ids) or (cp.isDeleted is false and d.datasetId IN UNNEST(@dataset_ids)))";
-    }
-    if (policyIds && policyIds.length > 0) {
-        filter = "(cp.policyId IN UNNEST(@policy_ids))";
-    }
-    else if (datasetIds && datasetIds.length > 0) {
-        filter = "(cp.isDeleted is false and d.datasetId IN UNNEST(@dataset_ids))";
-    }
-    else {
-        return;
+async function performTableMetadataUpdate(projectId, datasetId, tableId, accounts) {
+    const accessTypes = ["user", "group"];
+    const viewerRole = 'roles/bigquery.dataViewer';
+    console.log(`Begin metadata update for table: ${datasetId}.${tableId}`);
+    let isDirty = false;
+    const tablePolicy = await bigqueryUtil.getTableIamPolicy(projectId, datasetId, tableId);
+
+    let readBinding = {};
+    if (tablePolicy && tablePolicy.bindings) {
```

Comment:
Need to harden this logic still - if tablePolicy/bindings/members comes back undefined, need to handle.